### PR TITLE
tkt-42923: fix(vm/start): We now validate on start

### PR DESCRIPTION
**Pull request by skarekrow**
Ticket: #42923

### CODE REVIEW

File: src/middlewared/middlewared/plugins/vm.py [hunk 1]
```diff
@@ -1038,12 +1041,16 @@ def random_mac(self):
                         f'{schema_name}.vcpus',
                         'Only one Virtual CPU is allowed in this system.',
                     )
+            elif not flags['intel_vmx'] and not flags['amd_rvi']:
+                raise CallError(
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/vm.py]: I think CallError only makes sense for the vm.start here, for create/update it should raise a validation error

File: src/middlewared/middlewared/plugins/vm.py [hunk 2]
```diff
@@ -1125,14 +1132,19 @@ def random_mac(self):
...
         options.overcommit defaults to false, which means VM will not be allowed to
         start if there is not enough available memory to hold all VMs configured memory.
-        If true VM will start even if there is not enough memory for all VMs configured memory."""
-
+        If true VM will start even if there is not enough memory for all VMs configured memory.
+        """
         vm = await self._get_instance(id)
 
+        verrors = ValidationErrors()
+        await self.__common_validation(verrors, 'vm_start', vm)
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/vm.py]: It seems overkill to validatye all data again on vm.start? Perhaps we should just call vm.flags here and raise a CallError if not supported? Thoughts?

File: src/middlewared/middlewared/plugins/vm.py (hunk 1 above)
Comment (skarekrow) [reply src/middlewared/middlewared/plugins/vm.py]: This only happens during start, they wouldn't be able to create/update as the page wouldn't show up.

File: src/middlewared/middlewared/plugins/vm.py (hunk 2 above)
Comment (skarekrow) [reply src/middlewared/middlewared/plugins/vm.py]: Running through all of validation again ensures that everything is still correct, there could be a significant period of time between creation and start, including a bad database perhaps. This avoids a nasty set of issues later on, and doesn't take too much time

File: src/middlewared/middlewared/plugins/vm.py (hunk 1 above)
Comment (william-gr) [reply src/middlewared/middlewared/plugins/vm.py]: How? As far as I can see CallError will happen during create/update.

File: src/middlewared/middlewared/plugins/vm.py (hunk 2 above)
Comment (william-gr) [reply src/middlewared/middlewared/plugins/vm.py]: Except we are now using to validate code on a place not prepared to handle the exception we are now raising and making the validation code dirty with a CallError and schema name

File: src/middlewared/middlewared/plugins/vm.py [hunk 3]
```diff
@@ -1038,6 +1041,11 @@ def random_mac(self):
                         f'{schema_name}.vcpus',
                         'Only one Virtual CPU is allowed in this system.',
                     )
+            elif not flags['intel_vmx'] and not flags['amd_rvi']:
+                verrors.add(
+                    schema_name,
+                    'Virtualization is not supported by this system!'
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/vm.py]: Don't shout at the user.
```suggestion
                    'This system does not support virtualization.'
```

File: src/middlewared/middlewared/plugins/vm.py [hunk 4]
```diff
@@ -1125,13 +1133,19 @@ def random_mac(self):
...
-        If true VM will start even if there is not enough memory for all VMs configured memory."""
-
+        If true VM will start even if there is not enough memory for all VMs configured memory.
+        """
         vm = await self._get_instance(id)
+        flags = await self.middleware.call('vm.flags')
+
+        if not flags['intel_vmx'] and not flags['amd_rvi']:
+            raise CallError(
+                'Virtualization is not supported by this system!'
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/vm.py]:
```suggestion
                    'This system does not support virtualization.'
```

File: src/middlewared/middlewared/plugins/vm.py [hunk 5]
```diff
@@ -1038,6 +1041,11 @@ def random_mac(self):
                         f'{schema_name}.vcpus',
                         'Only one Virtual CPU is allowed in this system.',
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/vm.py]:
```suggestion
                        'Only one virtual CPU is allowed in this system.',
```

File: src/middlewared/middlewared/plugins/vm.py [hunk 6]
```diff
@@ -1125,13 +1133,19 @@ def random_mac(self):
...
     async def start(self, id, options):
-        """Start a VM.
-
+        """
+        Start a VM.
         options.overcommit defaults to false, which means VM will not be allowed to
         start if there is not enough available memory to hold all VMs configured memory.
-        If true VM will start even if there is not enough memory for all VMs configured memory."""
-
+        If true VM will start even if there is not enough memory for all VMs configured memory.
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/vm.py]:
```suggestion
        If true, VM starts even if there is not enough memory for all configured VMs.
```

File: src/middlewared/middlewared/plugins/vm.py [hunk 7]
```diff
@@ -1125,13 +1133,19 @@ def random_mac(self):
     @item_method
     @accepts(Int('id'), Dict('options', Bool('overcommit')))
     async def start(self, id, options):
-        """Start a VM.
-
+        """
+        Start a VM.
         options.overcommit defaults to false, which means VM will not be allowed to
         start if there is not enough available memory to hold all VMs configured memory.
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/vm.py]:
```suggestion
        start if there is not enough available memory to hold all configured VMs.
```

File: src/middlewared/middlewared/plugins/vm.py [hunk 8]
```diff
@@ -1125,13 +1133,19 @@ def random_mac(self):
     @item_method
     @accepts(Int('id'), Dict('options', Bool('overcommit')))
     async def start(self, id, options):
-        """Start a VM.
-
+        """
+        Start a VM.
         options.overcommit defaults to false, which means VM will not be allowed to
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/vm.py]:
```suggestion
        options.overcommit defaults to false, meaning VMs are not allowed to
```

**Pull request merged**